rubicon_ml/intake_rubicon/publish.py

```diff
@@ -18,13 +19,24 @@ def publish(
         and key to log the generated YAML file to. S3 buckets
         must be prepended with 's3://'. Defaults to None,
         which disables writing the generated YAML.
+    base_catalog_filepath : str, optional
+        Similar to output_filepath except this argument is used as a
+        base fileto update exisiting intake catalog. Defaults to None,
+        where the function proceeds as originally intended.
 
     Returns
     -------
     str
         The YAML string representation of the `intake` catalog
         containing the experiments `experiments`.
     """
+
+    if base_catalog_filepath is not None:
+        return update_catalog(
+            base_catalog_filepath=base_catalog_filepath,
+            new_experiments=experiments,
+            output_filepath=output_filepath,
+        )
```

Comment:
more nitpicking - can you add a newline after this. I'm surprised `black` didn't complain about that